api/server/router/distribution/distribution_routes.go

```diff
@@ -122,7 +122,7 @@ func (s *distributionRouter) getDistributionInfo(ctx context.Context, w http.Res
 		var platform v1.Platform
 		if err == nil {
 			err := json.Unmarshal(configJSON, &platform)
-			if err == nil {
+			if err == nil && (platform.OS != "" || platform.Architecture != "") {
```

Comment:
I'm thinking `OSVersion` and `OSFeatures` wouldn't make sense if `OS` is unset. But I can add these if you want.